include/taskolib/exceptions.h

```diff
@@ -48,57 +49,44 @@ namespace task {
  * {
  *     std::cerr << e.what() << "\n";
  * }
- * \endcode
  *
- * \note
- * task::Error is derived from std::runtime_error. It can therefore be caught by
- * `catch (const std::exception&)`, `catch (const std::runtime_error&)`, and
- * `catch (const task::Error&)`.
- */
-class Error : public std::runtime_error
-{
-public:
-    using std::runtime_error::runtime_error;
-};
-
-/**
- * An exception class storing the index of the step in which an error occurred, in
- * addition to the error message.
- *
- * \code
  * try
  * {
- *     throw task::ErrorAtIndex("An error has occurred", 42);
+ *     throw task::Error("An error has occurred", 42);
  * }
- * catch (const task::ErrorAtIndex& e)
+ * catch (const task::Error& e)
  * {
- *     std::cerr << e.what() << " in step " << e.get_index() << "\n";
+ *     std::cerr << e.what();
+ *     auto maybe_step_index = e.get_index();
+ *     if (maybe_step_index)
+ *         std::cerr << ": step index " << *maybe_step_index;
+ *     std::cerr << "\n";
  * }
  * \endcode
  *
  * \note
- * task::ErrorAtIndex inherits from std::runtime_error. It can therefore be caught by
- * `catch (const std::exception&)`, `catch (const std::runtime_error&)`,
- * `catch (const task::Error&)`, and `catch (const task::ErrorAtIndex&)`.
+ * task::Error is derived from std::runtime_error. It can therefore be caught by
+ * `catch (const std::exception&)`, `catch (const std::runtime_error&)`, and
+ * `catch (const task::Error&)`.
  */
-class ErrorAtIndex : public Error
+class Error : public std::runtime_error
 {
 public:
-    ErrorAtIndex(const std::string& msg, StepIndex index)
```

Comment:
By reducing the exception hierarchy for taskolib to only one generic `Error` by adding `StepIndex` means that we lose the ability to do so?

Reply:
Ok, this is a solution for now but I would expect that more fine granulated use cases will pop-up where we need to transfer more specialiced data.